graphql_compiler/tests/test_helpers.py

```diff
@@ -559,23 +558,26 @@ def compare_ignoring_whitespace(
     test_case.assertEqual(transform(expected), transform(received), msg=msg)
 
 
+def _lexicographic_sort_schema_text(schema_text: str) -> str:
+    """Sort the schema types and fields in a lexicographic order."""
+    return print_schema(lexicographic_sort_schema(build_schema(schema_text)))
+
+
 def compare_schema_texts_order_independently(
     test_case: TestCase, expected_schema_text: str, received_schema_text: str,
 ) -> None:
     """Compare expected and received schema texts, ignoring order of definitions."""
-    expected_schema_blocks = expected_schema_text.split("\n\n")
-    received_schema_blocks = expected_schema_text.split("\n\n")
-
-    # This is the preferred order-independent comparison method in Python:
-    # https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual
-    test_case.assertCountEqual(expected_schema_blocks, received_schema_blocks)
+    sorted_expected_schema_text = _lexicographic_sort_schema_text(expected_schema_text)
+    sorted_received_schema_text = _lexicographic_sort_schema_text(received_schema_text)
+    msg = "\n{}\n\n!=\n\n{}".format(sorted_expected_schema_text, sorted_received_schema_text)
+    compare_ignoring_whitespace(
+        test_case, sorted_expected_schema_text, sorted_received_schema_text, msg
```

Comment:
This PR is a really nice catch! but isn't `assertCountEqual` fine here, just needing to change the one line from `expected` to `received`?

Reply:
No. This function is splitting the schema text by lines and then comparing that the lines are the same. This does not work. The reason it doesn't is that we might split up the description of a directive/scalar into multiple lines in SCHEMA_TEXT so that we don't exceed the 100 character limit. The auto-generated schema doesn't split descriptions into multiples, (or at the very least doesn't use the same line separation we do).